cmd/catalyst-node/catalyst-node.go

```diff
@@ -348,3 +359,66 @@ func main() {
 
 	os.Exit(exitCode)
 }
+
+func startCatalystWebServer(httpAddr string) {
+	http.Handle("/hls/", redirectHlsHandler())
+	glog.Infof("HTTP server listening on %s", httpAddr)
+	http.ListenAndServe(httpAddr, nil)
+}
+
+var getClosestNode = queryMistForClosestNode
+
+func redirectHlsHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		playbackID, isValid := parsePlaybackID(r.URL.Path)
+		if !isValid {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		nodeAddr, err := getClosestNode(playbackID)
+		if err != nil {
+			glog.Error(err)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		redirectUrl := fmt.Sprintf("%s://%s/hls/%s/index.m3u8", protocol(r), nodeAddr, playbackID)
+		http.Redirect(w, r, redirectUrl, http.StatusFound)
+	})
+}
+
+func parsePlaybackID(path string) (string, bool) {
+	r := regexp.MustCompile("^/hls/([a-zA-Z0-9_+]+)/index.m3u8$")
```

Comment:
Dashes are valid too:
```suggestion
	r := regexp.MustCompile("^/hls/([a-zA-Z0-9_\-+]+)/index.m3u8$")
```